components/homme/src/preqx/share/prim_state_mod.F90

```diff
@@ -319,7 +317,7 @@ subroutine prim_printstate(elem, tl,hybrid,hvcoord,nets,nete)
 
     !   mass = integral( ps-p(top) )
     do ie=nets,nete
-       tmp(:,:,ie)=elem(ie)%state%ps_v(:,:,n0) 
+       tmp(:,:,ie)=hvcoord%hyai(1)*hvcoord%ps0 + sum(elem(ie)%state%dp3d(:,:,:,n0),3) 
```

Comment:
A general comment about all cases where surface pressure is defined in this way.  Isn't `hyai(1)` the top of the model?  Shouldn't it be `hyai(nlev)`?

Reply:
the full formula would be:
sum(dp) = [ hyai(nlevp)-hyai(1) ] p0  + [ hybi(nlevp)-hyba(1)] ps 

But at the surface, we must be pure terrain following and hyai(nlevp)=0, hybi(nlevp)=1
At the TOM, we must be pure pressure, and hybi(1)=0,  Thus we are left with:

sum(dp) = -hyai(1)*p0 + ps

hyai(1)*p0 is the pressure at the TOM  (prescribed as a boundary condition)

Reply:
ahhh yes, thank you for the explanation.